src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -134,7 +136,16 @@ internal static TimeSpan ResolveTimeSpanFromMilliseconds(string nameOfProperty,
 
         private static async Task CreateLeaseCollectionIfNotExistsAsync(CosmosClient cosmosClient, string databaseName, string collectionName, int? throughput)
         {
-            await CosmosDBUtility.CreateDatabaseAndCollectionIfNotExistAsync(cosmosClient, databaseName, collectionName, LeaseCollectionRequiredPartitionKey, throughput);
+            try
+            {
+                await CosmosDBUtility.CreateDatabaseAndCollectionIfNotExistAsync(cosmosClient, databaseName, collectionName, LeaseCollectionRequiredPartitionKey, throughput);
+            }
+            catch (CosmosException cosmosException) 
+                when (cosmosException.StatusCode == System.Net.HttpStatusCode.BadRequest
+                    && cosmosException.Message.Contains("invalid for Gremlin API"))
```

Comment:
Is this message guaranteed to stay the same? Wondering if there's any error code or something that would be safer to check.

Reply:
Sadly no :( The backend does not return a specific error code (Status Code + Sub status code). It's a plain HTTP 400, only way to detect it at this point is the error message.